transcom/mymove
# MB-15951: Add field to UI and office API

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15951)

## Summary

We are only showing the new section if the move the current SIT item corresponds to a destination SIT since this is only applicable for destination sit. 

Note the existing APIs that fetch mto_service_items for the office app do not contain the two newly added fields. Adding them was not necessary for this functionality, but likely something we want to add for consistency. This functionality was added as part of [this commit](https://github.com/transcom/mymove/pull/11311/commits/5ca75f10967fa07df9320991e5f2e125247aa325).

I didn't get a chance to validate if we want to make changes to any existing testharness data or for any playwright tests. They aren't necessary, but this is something we may want to include for future use.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the office app as a TOO using the pre-populated moveCode `DSTSIT`. View the Move task order page and see the default state
2. Login as a prime and view the move `DSTSIT`. Find the id of the DDDSIT service item and update it using postman. You can use the following information:
    - PATCH http://officelocal:3000/prime/v1/mto-service-items/{id}
    - 
         ```
        {
            "id": "<id here>",
            "modelType": "UpdateMTOServiceItemSIT",
            "sitCustomerContacted": "2023-08-31",
            "sitRequestedDelivery": "2023-09-02",
            "reServiceCode": "DDDSIT"
          }
         ```
3. You can then view the move as a TOO to view the populated fields

## Screenshots
Ignore the contents of the data. They are just showing the new sections. Storybook might be a good alternative for just viewing this as well.
![image](https://github.com/transcom/mymove/assets/84742904/56865256-a1b5-4ee6-8dd2-eeba7ae39860)
![image](https://github.com/transcom/mymove/assets/84742904/8618e85a-c06f-4a8b-be7f-b73e01aa11b7)
